core/vtype/src/main/java/org/phoebus/core/vtypes/VTypeHelper.java

```diff
@@ -506,4 +506,49 @@ public static boolean toBoolean(VType vtype) {
         }
         return false;
     }
+
+    /**
+     * Extracts the raw value from the given data object. The raw value is either one of the primitive wrappers or some
+     * kind of a list type if the value is an {@link Array}.
+     *
+     * @param type the value to extract the raw data from
+     * @return the raw data
+     */
+    public static Object toObject(VType type) {
+        if (type == null) {
+            return null;
+        }
+        if (type instanceof VNumberArray) {
+            if (type instanceof VIntArray || type instanceof VUIntArray) {
+                return VTypeHelper.toIntegers(type);
+            } else if (type instanceof VDoubleArray) {
+                return VTypeHelper.toDoubles(type);
+            } else if (type instanceof VFloatArray) {
+                return VTypeHelper.toFloats(type);
+            } else if (type instanceof VLongArray || type instanceof VULongArray) {
+                return VTypeHelper.toLongs(type);
+            } else if (type instanceof VShortArray || type instanceof VUShortArray) {
+                return VTypeHelper.toShorts(type);
+            } else if (type instanceof VByteArray || type instanceof VUByteArray) {
+                return VTypeHelper.toBytes(type);
+            }
+        } else if (type instanceof VEnumArray) {
```

Comment:
It's inconsistent to convert a VEnumArray to List<String>, but a single enum is converted to its index.
With an enum having both a numeric and a string value, it's of course a good question which one to pick, and I'm not sure how you want to decide that right now, but the result should be consistent.

Reply:
@georgweiss I imagine in both cases it should be an int?

Reply:
Int has the advantage that you can plot it over time. But you never know if 0/1 represents OK/Fault or Off/On or Left/Right etc. Still, turning it into a number is likely best.

Reply:
Would be nice to have a feature in the plot which could automatically display the enum value.